bin/picca_Pk1D.py

```diff
@@ -168,7 +176,15 @@ def process_all_files(index_file_args):
                      pixel_step, log_lambda_array[part_index],
                      delta_array[part_index], exposures_diff_array[part_index],
                      ivar_array[part_index], args.no_apply_filling)
-            if num_masked_pixels > args.nb_pixel_masked_max:
+            
+            if (args.nb_pixel_masked_max is not None):
+                max_num_masked_pixels = args.nb_pixel_masked_max
+            elif linear_binning:
+                max_num_masked_pixels = int(args.nb_pixel_masked_frac_max*(np.max(lambda_new)-np.min(lambda_new))/pixel_step)           #this only accounts for masking inside the spectrum, not at the adges
+            else:
+                max_num_masked_pixels = int(args.nb_pixel_masked_frac_max*(np.max(log_lambda_new)-np.min(log_lambda_new))/pixel_step)
+                            
+            if num_masked_pixels > max_num_masked_pixels:
```

Comment:
So now the default is the fraction right ? but you can still force a mode with a number of pixels as previously.

Reply:
the idea is: you can give a fraction or number via the appropriate keywords, if you give both the code prints a message and uses the fraction, if you give none, the old default pixel number is used, elsewise it uses whatever you tell it to use...